src/WorldStorage/FastNBT.cpp

```diff
@@ -503,7 +503,7 @@ void cFastNBTWriter::AddString(const AString & a_Name, const AString & a_Value)
 void cFastNBTWriter::AddByteArray(const AString & a_Name, const char * a_Value, size_t a_NumElements)
 {
 	TagCommon(a_Name, TAG_ByteArray);
-	u_long len = htonl(static_cast<u_long>(a_NumElements));
+	u_int len = htonl(static_cast<u_int>(a_NumElements));
```

Comment:
And on windows u_long is identical to u_int, so the cast doesn't matter, whereas on Linux its a uint32_t, which is a unsigned int, and is different from a long.